crates/tauri-bundler/src/bundle.rs

```diff
@@ -126,7 +126,9 @@ pub fn bundle_project(settings: &Settings) -> crate::Result<Vec<Bundle>> {
       continue;
     }
 
-    patch_binary(&settings.binary_path(main_binary), package_type)?;
+    if let Err(e) = patch_binary(&settings.binary_path(main_binary), package_type) {
+      log::warn!("Failed to add package type information to the binary. Updater plugin may not be able to update this package: {}", e);
```

Comment:
```suggestion
      log::warn!("Failed to add bundle type to the binary: {e}. Updater plugin may not be able to update this package. This shouldn't normally happen, please report it to https://github.com/tauri-apps/tauri/issues");
```

We should let people know that this is probably not their fault if it failed and let them report the problem